ethtool: Hide unsupported ethtool features during querying

cathay4t opened the pull request:
Only provide NetworkManager supported ethtool features during querying.

Integration test case included which require linux kernel 6.15+.

ihuguet commented on rust/src/lib/nispor/ethtool.rs:
```diff
@@ -21,7 +82,20 @@ fn gen_ethtool_config(ethtool_info: &nispor::EthtoolInfo) -> EthtoolConfig {
+                .changeable
+                .iter()
+                .filter_map(|(name, value)| {
+                    if SUPPORTED_ETHTOOL_FEATURES.contains(&name.as_str()) {
```
As the features list is sorted, could we use binsearch?

cathay4t replied:
If follow up developer forgot to sort it, this binsearch will cause problem.

Let me generate hashset at runtime, it is a small cost to generated and should be faster than binsearch also.

cathay4t replied:
Changed to `HashSet` and also added comment: `TODO: Use OnceCell`. I don't time for this low priority fix, maybe later.

vbenes commented:
tested on F42, working well, just one error in test_iface_down_with_routes_in_current[ipv6] but that is very likely unrelated.

ihuguet commented on rust/src/lib/nispor/ethtool.rs:
```diff
@@ -20,8 +81,24 @@ fn gen_ethtool_config(ethtool_info: &nispor::EthtoolInfo) -> EthtoolConfig {
         ret.pause = Some(pause_config);
     }
+    // TODO(Gris Ge): Use `OnceCell`
+    let supported_features: HashSet<&&str> =
+        SUPPORTED_ETHTOOL_FEATURES.iter().collect();
```
I think you can avoid the `&&` using `iter().copied()` or `cloned()`?

cathay4t replied:
`&&` is better than clone action.
Never mind, let me fix it for good with OnceCell.

cathay4t replied:
Changed to `copied()` and also using OnceLock to prevent second creation of HashSet.

cathay4t committed:
ethtool: Hide unsupported ethtool features during querying

Only provide NetworkManager supported ethtool features during querying.

Integration test case included which require linux kernel 6.15+.

Signed-off-by: Gris Ge <fge@redhat.com>
